### DISCUSSION TITLE
"Chromedriver not in PATH" error

### DISCUSSION THREAD

**Discussion by JackBlackburnBuffs** [gramineproject/gramine#1231] [Q&A] [answered]
I'm currently trying to use gramine-sgx to run a python script that has a set list of URLs that uses selenium chromedriver to make requests to these URLs but I'm getting a "selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://chromedriver.chromium.org/home" error as I run the script. I know it's in the path because when I check the environment variable has /usr/bin in the path which is where the executable is stored. So I'm wondering is this an issue with the gramine side of things?? I'm new to gramine so any help is appreciated.

**Comment by kailun-qin** [comment, marked as answer]
Hi @JackBlackburnBuffs, thanks for your question!

Gramine-SGX by default does not pass environment variables (e.g., `PATH`) from the host to the app. Thus you need to specifiy them explicitly in your Gramine manifest file. Pls kindly see the detailed options here: https://gramine.readthedocs.io/en/stable/manifest-syntax.html#environment-variables.

**Reply by JackBlackburnBuffs** [reply]
Thank you for your reply, I looked into that and it seems this is exactly what I need. However, I'm struggling to get this to work. I've added:

loader.env.[ENVIRON] = {passthrough = true}
loader.env_src_file = "file:/usr/bin/chromedriver"

to the manifest template file but when I try to make the project I'm getting this error:
"
tomli.TOMLDecodeError: Invalid initial character for a key part (at line 8, column 12)
make: *** [Makefile:20: python.manifest] Error 1 
"
I've tried all the different options that it gives to pass environment variables but it still gives me this error. How should I move forward from here?

**Reply by monavij** [reply]
I suggest first getting you app running by using loader.insecure__use_host_env = [true]

Did you generate your enc_src_file using Gramine tool [gramine-argv-serializer]?

**Reply by dimakuv** [reply]
@JackBlackburnBuffs Please don't read the documentation so literally :)

What you need to do is to add `loader.env.PATH = "/usr/bin"`.

You **don't** need to add `loader.env_src_file = "file:/usr/bin/chromedriver"` (please read carefully the docs, this manifest option is for a specially-prepared file with environment variables listed in it).

You **can't** add `loader.env.[ENVIRON] = {passthrough = true}`, because in our documentation `[ENVIRON]` means "put the name of the environment variable that you want to pass-through into Gramine".

**Reply by JackBlackburnBuffs** [reply]
@dimakuv  @monavij Thank you for your responses, Dmitrii was correct, and I've now solved the error. I figured that might be the issue, so thank you for clarifying.
